avalanche/models/timm_vit.py

```diff
@@ -0,0 +1,1118 @@
+""" Vision Transformer (ViT) in PyTorch
+A PyTorch implement of Vision Transformers as described in:
+'An Image Is Worth 16 x 16 Words: Transformers for Image Recognition at Scale'
+    - https://arxiv.org/abs/2010.11929
+`How to train your ViT? Data, Augmentation, and Regularization in Vision 
+Transformers`
+    - https://arxiv.org/abs/2106.10270
+The official jax code is released and available at: 
+https://github.com/google-research/vision_transformer
+
+Acknowledgments:
+* The paper authors for releasing code and weights, thanks!
+* I fixed my class token impl based on Phil Wang's 
+https://github.com/lucidrains/vit-pytorch ... check it out
+for some einops/einsum fun
+* Simple transformer style inspired by Andrej Karpathy's: 
+https://github.com/karpathy/minGPT
+* Bert reference code checks against Huggingface Transformers and 
+Tensorflow Bert
+Hacked together by / Copyright 2020, Ross Wightman
+# ------------------------------------------
+# Based on code added by:
+# -- Jaeho Lee, dlwogh9344@khu.ac.kr
+# ------------------------------------------
+"""
+import logging
+import math
+from collections import OrderedDict
+from functools import partial
+import re
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import torch.utils.checkpoint
+from avalanche.models import Prompt
+
+try:
+    from timm.data import (
+        IMAGENET_DEFAULT_MEAN,
+        IMAGENET_DEFAULT_STD,
+        IMAGENET_INCEPTION_MEAN,
+        IMAGENET_INCEPTION_STD,
+    )
+    from timm.models.helpers import (
+        adapt_input_conv,
+        build_model_with_cfg,
+        checkpoint_seq,
+        named_apply,
+        resolve_pretrained_cfg,
+    )
+    from timm.models.layers import DropPath, Mlp, PatchEmbed, \
+        lecun_normal_, trunc_normal_
+    from timm.models.registry import register_model
+
+except ImportError:
+    raise ModuleNotFoundError(
+        "timm is not found, "
+        "if you want to use L2P "
+        "please install avalanche with the "
+        "detection dependencies: "
+        "pip install avalanche-lib[l2p]"
+    )
+
+_logger = logging.getLogger(__name__)
+
+
+def _cfg(url="", **kwargs):
+    return {
+        "url": url,
+        "num_classes": 1000,
+        "input_size": (3, 224, 224),
+        "pool_size": None,
+        "crop_pct": 0.9,
+        "interpolation": "bicubic",
+        "fixed_input_size": True,
+        "mean": IMAGENET_INCEPTION_MEAN,
+        "std": IMAGENET_INCEPTION_STD,
+        "first_conv": "patch_embed.proj",
+        "classifier": "head",
+        **kwargs,
+    }
+
+
+default_cfgs = {
```

Comment:
do we need these configs in avalanche? aren't they already available on timm? I would avoid copy-pasting things that people can take from somewhere else.